src/pyhf/tensor/numpy_backend.py

```diff
@@ -1,60 +1,85 @@
 """NumPy Tensor Library Module."""
+from __future__ import annotations
 import numpy as np
 import logging
 from scipy.special import gammaln, xlogy
 from scipy import special
 from scipy.stats import norm, poisson
 
+from pyhf.typing import Shape, Literal
+
+from typing import TypeVar, Callable, Sequence, Generic, Mapping, Union, TYPE_CHECKING
+from numpy.typing import (
+    NDArray,
+    NBitBase,
+    ArrayLike,
+    DTypeLike,
+)
+
+T = TypeVar("T", bound=NBitBase)
+
+if TYPE_CHECKING:
+    Tensor = Union[NDArray[np.number[T]], NDArray[np.bool_]]
+else:
+    Tensor = Generic
```

Comment:
It's much easier to find these and fix (remove) them later, e.g. with `pyupgrade` presumably, then if they were wrapped in strings. Why is this an issue though? This is why the `TYPE_CHECKING` was added.

Reply:
I guess it isn't an Issue, but @henryiii has pointed out that that they aren't necessary in https://github.com/scikit-hep/pyhf/pull/1909#discussion_r944456765. I'm not expecting `pyupgrade` to fix them, but I'll be impressed if it does. :+1: @kratsg Though if this is a Python version specific Issue then please make a comment about what version of Python we need to drop to get this, like

https://github.com/scikit-hep/pyhf/blob/f01dcf59da10ec9313345305efc49cfb7186bf99/src/pyhf/typing.py#L30-L31

so that it is easy to find them when `git grep`ing on Python `3.x` numbers to see what needs updating if it has to be done manually.